### PR TITLE
Builtin source

### PR DESCRIPTION
**Pull request by GitBluub**
two branches at once because I didn't go back to master to do it ><, builtin source and builtin which/where. to add aliases when they are added to the shell. Naming of functions may still be bad.

### CODE REVIEW

**Review by zoriya — CHANGES_REQUESTED** [review]
Things that need to be changed:
- You don't check your mallocs (ex: the start_script function).
- You should use functions name that won't conflict with other files or make them statics. Ex: start_script is a name that could be used for anything.

The rest is good !

PS: The source branch should only contain the source built-in and not the where and which since they are not merged with master yet.

**Review by zoriya — APPROVED** [review]
Good. I think we should wait for the where/which to be merged with master before merging this one but we are good.

**Review by melefo — APPROVED** [review]

**Review by Octopus773 — APPROVED** [review]

**Review by Arthi-chaud — APPROVED** [review]

**Comment by zoriya** [comment]
That seems good to me but keep in mind that $1, $2, $@ etc should be handled like other variables so this should be changed when the parser will handle variables.

**Review by Arthi-chaud — APPROVED** [review]

**Pull request merged**
